src/main/java/org/spongepowered/api/service/permission/SubjectCollection.java

```diff
@@ -24,86 +24,315 @@
+     * data is not considered.</p>
      *
-     * <p>Note: This data may be persisted, so plugins that add permissions to
-     * this subject must take care to not override permissions already set or
-     * modified. It is also recommended to use
-     * {@link Subject#getTransientSubjectData()} where possible to
-     * avoid persisting unnecessary data.
+     * <p>As no context is passed, it is up to the implementation to decide
+     * which contexts to use. When available,
+     * {@link Subject#getActiveContexts()} is used for the lookup. Otherwise, it
+     * is likely that {@link SubjectData#GLOBAL_CONTEXT} will be
+     * used.</p>
      *
-     * @return The subject holding defaults
+     * @param permission The permission to check
+     * @return A map containing any subject known to have this permission set,
+     *         and the value this permission is set to
+     */
+    Map<Subject, Boolean> getLoadedWithPermission(String permission);
+
+    /**
+     * Return all loaded subjects with the given permission set.
+     *
+     * <p>This method <p>DOES NOT</p> consider inheritance, and will only query
+     * the data set to the subjects {@link Subject#getSubjectData()}. Transient
+     * data is not considered.</p>
+     *
+     * @param contexts The context combination to check for permissions in
+     * @param permission The permission to check
+     * @return A map containing any subject known to have this permission set,
+     *         and the value this permission is set to
+     */
+    Map<Subject, Boolean> getLoadedWithPermission(Set<Context> contexts, String permission);
+
+    /**
+     * Gets the subject holding data that is applied by default to all
+     * subjects in this collection.
+     *
+     * <p>This subject is at the root of all inheritance trees for subjects in
+     * this collection, but at a higher priority chan defaults expressed to
+     * {@link PermissionService#getDefaults()}.</p>
+     *
+     * <p>Note: This data may be persisted, so plugins that add
+     * permissions to this subject must take care to not override
+     * permissions already set or modified.</p>
+     *
+     * <p>It is also recommended to use
+     * {@link Subject#getTransientSubjectData()} where possible to avoid
+     * persisting unnecessary data.</p>
+     *
+     * <p>Assigning default permissions should be used sparingly, and by
+     * convention, only in situations where "default" game behaviour is restored
+     * by granting a certain permission.</p>
+     *
+     * @return The subject holding defaults for this collection
      */
     Subject getDefaults();
+
+    /**
+     * Indicate that a certain subject may be unloaded by the implementation.
+     *
+     * <p>This is only a hint to the implementation, and does not guarantee that
+     * the subject will be unloaded.</p>
+     *
+     * @param identifier The subject to be unloaded
+     */
+    void suggestUnload(String identifier);
```

Comment:
> Is it ever imperative to know if the implementation actually did it?

I don't think so. I'd extend that to say it would never even be useful, let alone imperative.